impacket/dcerpc/v5/dhcpm.py

```diff
@@ -145,10 +145,13 @@ class DHCP_CLIENT_INFO_VQ(NDRSTRUCT):
     )
 
 class DHCP_CLIENT_SEARCH_UNION(NDRUNION):
+    commonHdr = (
+        ('tag', DHCP_SEARCH_INFO_TYPE),
+    )
     union = {
-        DHCP_SEARCH_INFO_TYPE.DhcpClientIpAddress:       ('ClientIpAddress', DHCP_IP_ADDRESS),
-        DHCP_SEARCH_INFO_TYPE.DhcpClientHardwareAddress: ('ClientHardwareAddress', DHCP_CLIENT_UID),
-        DHCP_SEARCH_INFO_TYPE.DhcpClientName:            ('ClientName', LPWSTR),
+        DHCP_SEARCH_INFO_TYPE.enumItems.DhcpClientIpAddress.value:       ('ClientIpAddress', DHCP_IP_ADDRESS),
+        DHCP_SEARCH_INFO_TYPE.enumItems.DhcpClientHardwareAddress.value: ('ClientHardwareAddress', DHCP_CLIENT_UID),
+        DHCP_SEARCH_INFO_TYPE.enumItems.DhcpClientName.value:            ('ClientName', LPWSTR),
```

Comment:
Why are you defining the NDRUNION this way? it is cleaner to do it the way you used to do it before.
[Here](https://github.com/CoreSecurity/impacket/blob/master/impacket/dcerpc/v5/samr.py#L656) another example of similar behavor.